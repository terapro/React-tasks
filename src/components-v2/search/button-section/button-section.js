import React from 'react';
import PropTypes from 'prop-types';
import {Button} from 'src/components-v2/common/button/button.js';

export const ButtonSection = ({searchButtonClick}) => (
  <div>
    <Button content='Find it' btnClick={searchButtonClick} id={'search-1'} btnRed btnBig/>
  </div>
);

ButtonSection.propTypes = {
  searchButtonClick: PropTypes.func.isRequired
};
