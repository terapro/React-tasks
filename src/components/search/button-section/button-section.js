import React from 'react';
import PropTypes from 'prop-types';
import {FormButton} from '../../common/form-button/form-button.js';

const ButtonSection = (props) => (
    <div className={'search-button-section'}>
        <FormButton content={'Search'} parentFormId = {props.parentFormId}/>

    </div>
);

ButtonSection.propTypes = {
    parentFormId: PropTypes.string
};

export {ButtonSection};