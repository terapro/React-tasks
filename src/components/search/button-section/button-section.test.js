import React from 'react';
import {ButtonSection} from 'src/components/search/button-section/button-section';
import {shallow} from 'enzyme';

const mockFunc = jest.fn();

describe('<ButtonSection />', () => {
  describe('Render', () => {
    it('should render the component', () => {
      const wrapper = shallow(<ButtonSection searchButtonClick={mockFunc}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});