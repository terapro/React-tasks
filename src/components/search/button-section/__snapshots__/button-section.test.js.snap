// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonSection /> Render should render the component 1`] = `
<div>
  <Button
    btnBig={true}
    btnBold={false}
    btnClick={[MockFunction]}
    btnGrey={false}
    btnRed={true}
    btnSm={false}
    btnTransp={false}
    btnTranspRed={false}
    btnUC={false}
    btnWhite={false}
    content="Search"
    id=""
  />
</div>
`;
